internal/envelope/envelope.go

```diff
@@ -23,3 +43,37 @@ func GetEnvelopeMediaType(sigFormat string) (string, error) {
 	}
 	return "", fmt.Errorf("signature format %q not supported", sigFormat)
 }
+
+// DescriptorFromPayload parses a signature payload and returns the descriptor
+// that was signed. Note: the descriptor was signed but may not be trusted
+func DescriptorFromSignaturePayload(payload *signature.Payload) (*ocispec.Descriptor, error) {
+	if payload == nil {
+		return nil, errors.New("empty payload")
+	}
+
+	var intPayload intermediatePayload
+	err := json.Unmarshal(payload.Content, &intPayload)
+	if err != nil {
+		return nil, errors.New("failed to unmarshall the payload content to intermediatePayload")
+	}
+
+	validMediaType := false
+	for _, mediaType := range supportedMediaTypes {
+		if intPayload.TargetArtifact.MediaType == mediaType {
+			validMediaType = true
+			break
+		}
+	}
+
+	if !validMediaType {
+		return nil, errors.New(fmt.Sprintf("unsupported media type %s", intPayload.TargetArtifact.MediaType))
+	}
```

Comment:
Looks like you are trying to validate the `TargetArtifact.MediaType`. Should we instead validate the `payload.ContentType`? To validate the content type: https://github.com/notaryproject/notation-go/blob/1b2ec27aea7b2565b9adfb8d5d32e6df88ef4325/internal/envelope/envelope.go#L19. 
(Since it's unexported in notation-go, you could add it in internal/envelope/envlope.go at the CLI side as well.)

Reply:
the original comment that this was added for was specifically about validating the Payload MediaType ([link](https://github.com/notaryproject/notation-go/pull/261#discussion_r1098062259)), but i may have also misunderstood the ask there. can clarify whether we still want to validate the media type of the target artifact, but otherwise will add the helper for validating the payload content type

Reply:
> the original comment that this was added for was specifically about validating the Payload MediaType ([link](https://github.com/notaryproject/notation-go/pull/261#discussion_r1098062259)), but i may have also misunderstood the ask there. can clarify whether we still want to validate the media type of the target artifact, but otherwise will add the helper for validating the payload content type

We should always validate the `payload.ContentType` based on our [spec](https://github.com/notaryproject/notaryproject/blob/main/specs/signature-specification.md#payload). (Note in the example of the spec, the `payload.TargetArtifact` MediaType can be `application/vnd.oci.image.manifest.v1+json` or `sbom/example`)

Reply:
confirmed with shiwei that the comment was intended to be for ContentType, will remove the media type check